### ISSUE TITLE
[Annotation] create more informative exception for miss-use of @link/@see

### ISSUE THREAD

**Issue by TomasVotruba** [ApiGen/ApiGen#936]
Ref https://github.com/ApiGen/ApiGen/issues/912#issuecomment-313665106

**Comment by nelson6e65** [comment]
I think throwing exception for `@see` and `@link`  this is _too many_. Maybe a warning instead, because this are minor issues.

**Comment by TomasVotruba** [comment]
See this issue https://github.com/ApiGen/ApiGen/issues/864#issuecomment-305650637 for more. This is issue in phpDocumentor\TypeResolver package. Report it there, if you need to change this.

**Comment by nelson6e65** [comment]
I can change annotations in my code, but I can't do that for my vendor. In my case, is in my vendor referenced class that `@see` includes an url and throws this exception.

**Comment by soren121** [comment]
@TomasVotruba This is already an open issue in phpDocumentor/ReflectionDocBlock#78 and phpDocumentor/ReflectionDocBlock#88, but work has stalled.

Also, the PHPDoc team considers `@link` to be deprecated and prefer for `@see` to be used for both classes and external links. This is the opposite of what ApiGen is advocating right now.

**Comment by TomasVotruba** [comment]
@nelson6e65 Vendor should not be used directly, but I see that extending a class from vendor can cause this. @soren121 Thanks for your valuable insights both of you.

Do you have any temporary solution for this in mind? Probably around [`transform()` method](https://github.com/ApiGen/ApiGen/blob/6aa8eebde6e381f5f00bab16bc3760e9a6ebbe4e/packages/Reflection/src/TransformerCollector.php#L69).

**Comment by soren121** [comment]
Ideally, I think that in the event of any recoverable parsing error (not just for `@see` and `@link`), ApiGen should print a warning and continue generating the next part of the documentation. For users that don't want warnings, you could add a command-line argument that forces ApiGen to stop on the first warning, which would be the same as the current functionality.

This is how the popular [Sphinx](http://www.sphinx-doc.org/) doc generator currently works. I don't know how feasible this approach would be.

**Comment by TomasVotruba** [comment]
Thanks for the idea. In my experience, warnings tend to become white noise ignored by end user.

For 1st version, I think it would be just fine if it works. Without errors and command options. 
Would you be open such version and prepare PR for that?

**Comment by nelson6e65** [comment]
Or...

Generate a report at the end of `generate` like PHPUnit does for uncompleted and skipped tests.

I think is good that user should decide if wants to ignore or not this warnings via option in yml configuration file and/or command option.

And IMHO, at least for this concern  stop API because `@see` is too invasive.

**Comment by TomasVotruba** [comment]
@nelson6e65 I'm open to this. Would you prepare a PR?

**Comment by nelson6e65** [comment]
Yes, I can.

I tried an `bottle opener` for this kind of feature in #922. But I didn't know _where_ to put this `success`/`failed` message. So I put them in the [`__destructor` of `src/Console/Helper/ProgressBar.php`](https://github.com/ApiGen/ApiGen/pull/922/files#diff-80d6751078f273130251e16ea14e7cdfR46).

I think this should be made in parts, and finally include this kind of warnings report.

**Comment by TomasVotruba** [comment]
I linked the responsible method above: https://github.com/ApiGen/ApiGen/issues/936#issuecomment-313845332

If you manage to do working wrapper around it, we've 80 % work done.

**Comment by nelson6e65** [comment]
Well... I'm not familiar with that and... most of code is not documented. So, idk what `transform()` method actually does. :sweat_smile:

**Comment by TomasVotruba** [comment]
Sure. Basically it takes reflection from [BetterReflection](https://github.com/Roave/BetterReflection) - class reflection, trait reflection, property reflection etc.... an transforms it into [Apigen reflection](https://github.com/ApiGen/ApiGen/tree/master/packages/Reflection/src/Reflection) alternative.

The goal is to find where exactly is @see/@link exception in this issue risen and wrap around it.

```php
try {
    $transformedReflection = $transformer->transform($reflection);
} catch (Throwable $throwable) {
    // do some fallback magic here
}
```

And actually **a failing test would be the best to start with**.

**Comment by nelson6e65** [comment]
I need some :sleeping: for now.

I'll check later in the dawn. :coffee:

**Comment by TomasVotruba** [comment]
Same here, enjoy the :sleeping:

**Comment by TomasVotruba** [comment]
@see @link is probably caused by DocBlockFactory here: https://github.com/ApiGen/ApiGen/blob/master/packages/Reflection/src/Transformer/BetterReflection/Class_/ClassReflectionTransformer.php#L54

To-do: use own wrapper for DocBlockFactory

**Comment by TomasVotruba** [comment]
Allright, I took some time to dig-in this and the issue was located in this [`See` class](https://github.com/phpDocumentor/ReflectionDocBlock/blob/7c33776cd8f801a1f727b95adf714d465c9f523d/src/DocBlock/Tags/See.php#L59).

We used that before [BetterReflection construction in Transformers](https://github.com/ApiGen/ApiGen/blob/6a29177a0145de43dec2958797657fec386425c6/packages/Reflection/src/Transformer/BetterReflection/Class_/ClassReflectionTransformer.php#L54). So it has nothing to do 
with BetterReflection, as far as I know now. Sorry about wrong direction.

Own implementation of `See` class annotation that have 2 paths for `@see http://link.com` and `@see Class` should resolve this:
#952

Please review

**Comment by TomasVotruba** [comment]
Closed by #952
